raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -1,122 +1,108 @@
 
-    if event_type == "create_channel":
-        await create_channel(rtc_partner, ag_transceiver, partner_address, node_address)
-    if event_type == "message":
-        send_message(rtc_partner, event_data, node_address)
-    if event_type == "set_remote_description":
-        await set_remote_description(rtc_partner, ag_transceiver, event_data, node_address)
+def wait_for_future(future: Any, callback: Callable, partner_address: Address) -> None:
+    result = yield_future(future)
+    if callback is not None:
+        callback(result, partner_address)
 
 
 async def create_channel(
-    rtc_partner: RTCPartner, ag_transceiver: AGTransceiver, partner_address: Address, node_address
-) -> None:
+    peer_connections: Dict[Address, RTCPartner],
+    partner_address: Address,
+    node_address: Address,
+    handle_message_callback: Callable,
+) -> RTCSessionDescription:
+
+    if partner_address not in peer_connections:
+        peer_connections[partner_address] = RTCPartner(partner_address, RTCPeerConnection(), None)
+
+    rtc_partner = peer_connections[partner_address]
     pc = rtc_partner.pc
-    rtc_partner.create_channel()
+    rtc_partner.create_channel(node_address)
 
     offer = await pc.createOffer()
     await pc.setLocalDescription(offer)
 
-    event = {
-        "type": "local_description",
-        "data": {"sdp": pc.localDescription.sdp, "type": pc.localDescription.type},
-        "address": partner_address,
-    }
-
-    await ag_transceiver.send_event_to_gevent(event)
+    msg = "Channel must be created already"
+    assert rtc_partner.channel is not None, msg
 
     @rtc_partner.channel.on("message")
-    async def on_message(message: [str, bytes]) -> None:  # pylint: disable=unused-variable
+    async def on_message(message: Dict[str, bytes]) -> None:
         log.debug(
             "Received message in aio kingdom",
             node=to_checksum_address(node_address),
             message=message,
             time=time.time(),
         )
-        await ag_transceiver.send_event_to_gevent(
-            {"type": "message", "data": message, "address": partner_address}
-        )
+        handle_message_callback({"type": "message", "data": message, "address": partner_address})
+
+    return pc.localDescription
 
 
 async def set_remote_description(
-    rtc_partner: RTCPartner, ag_transceiver: AGTransceiver, description, node_address
-) -> None:
+    rtc_partner: RTCPartner,
+    description: Dict[str, str],
+    node_address: Address,
+    handle_message_callback: Callable,
+) -> Optional[RTCSessionDescription]:
+
     remote_description = RTCSessionDescription(description["sdp"], description["type"])
     sdp_type = description["type"]
     pc = rtc_partner.pc
     await pc.setRemoteDescription(remote_description)
 
     @rtc_partner.pc.on("datachannel")
-    def on_datachannel(channel):  # pylint: disable=unused-variable
+    def on_datachannel(channel: RTCDataChannel) -> None:  # pylint: disable=unused-variable
```

Comment:
Are the pylint ignores required?

Reply:
Yes. It complaints that it isnt used, but with the decorator it is attached to be a callback in the aiortc library.